src/webui/webapplication.cpp

```diff
@@ -893,98 +823,133 @@ void WebApplication::action_command_setCategory()
         BitTorrent::TorrentHandle *const torrent = BitTorrent::Session::instance()->findTorrent(hash);
         if (torrent) {
             if (!torrent->setCategory(category)) {
-                status(400, "Incorrect category name");
+                status(404, "Not Found");
+                print(tr("Incorrect category name"), Http::CONTENT_TYPE_TXT);
                 return;
             }
         }
     }
 }
 
-void WebApplication::action_command_addCategory()
+void WebApplication::addCategory()
 {
-    CHECK_URI(0);
     CHECK_PARAMETERS("category");
 
     QString category = request().posts["category"].trimmed();
 
     if (!BitTorrent::Session::isValidCategoryName(category) && !category.isEmpty()) {
-        status(400, tr("Incorrect category name"));
+        status(404, "Not Found");
+        print(tr("Incorrect category name"), Http::CONTENT_TYPE_TXT);
         return;
     }
 
     BitTorrent::Session::instance()->addCategory(category);
 }
 
-void WebApplication::action_command_removeCategories()
+void WebApplication::removeCategories()
 {
-    CHECK_URI(0);
     CHECK_PARAMETERS("categories");
 
     QStringList categories = request().posts["categories"].split('\n');
     foreach (const QString &category, categories)
         BitTorrent::Session::instance()->removeCategory(category);
 }
 
-void WebApplication::action_command_getSavePath()
+void WebApplication::getDefaultSavePath()
 {
-    CHECK_URI(0);
     print(BitTorrent::Session::instance()->defaultSavePath());
 }
 
-bool WebApplication::isPublicScope()
+bool WebApplication::isAPIMethod(const QString &methodName) const
 {
-    return (scope_ == DEFAULT_SCOPE || scope_ == VERSION_INFO);
+    return (m_privateAPIs.contains(methodName)
```

Comment:
m_publicAPIs list is shorter. Why don't you test it first?

Reply:
The public paths requested much less frequently. Don't forget that `syncMaindata` (private path) can be requested very frequently.

Reply:
OK.